Menus/SpatialUI/SpatialMenu/Scripts/SpatialMenuSubMenuElement.cs

```diff
@@ -0,0 +1,281 @@
+            if (selected != null)
+                selected(selectionNode);
+        }
+
+        public void SetupInternal(Transform parentTransform, Action selectedAction, String displayedText = null, string toolTipText = null)
+        {
+            if (selectedAction == null)
+            {
+                ObjectUtils.Destroy(gameObject);
+                return;
+            }
+
+            this.selectedAction = selectedAction;
+            m_RectTransform = (RectTransform)transform;
+            m_OriginalSize = m_RectTransform.sizeDelta;
+            m_ExpandedTooltipDisplaySize = new Vector2(m_RectTransform.sizeDelta.x, m_ExpandedTooltipHeight);
+
+            m_Icon.gameObject.SetActive(false);
+            m_Text.gameObject.SetActive(true);
+            m_Text.text = displayedText;
+
+            transform.SetParent(parentTransform);
+            transform.localRotation = Quaternion.identity;
+            transform.localPosition = Vector3.zero;
+            transform.localScale = Vector3.one;
+
+            if (Mathf.Approximately(m_TransitionDuration, 0f))
+                m_TransitionDuration = 0.001f;
+
+            // Tooltip text related
+            m_TooltipVisualsCanvasGroup.alpha = 0;
+            m_TooltipText.text = toolTipText;
+        }
+
+        IEnumerator AnimateVisibility(bool fadeIn)
+        {
+            var currentAlpha = fadeIn ? 0f : m_CanvasGroup.alpha;
+            var targetAlpha = fadeIn ? 1f : 0f;
+            var alphaTransitionAmount = 0f;
+            var textTransform = m_Text.transform;
+            var textCurrentLocalPosition = textTransform.localPosition;
+            textCurrentLocalPosition = fadeIn ? new Vector3(m_TextOriginalLocalPosition.x, m_TextOriginalLocalPosition.y, m_FadeInZOffset) : textCurrentLocalPosition;
+            var textTargetLocalPosition = m_TextOriginalLocalPosition;
+            var positionTransitionAmount = 0f;
+            var transitionSubtractMultiplier = 1f / m_TransitionDuration;
+            while (alphaTransitionAmount < 1f)
+            {
+                var alphaSmoothTransition = MathUtilsExt.SmoothInOutLerpFloat(alphaTransitionAmount);
+                var positionSmoothTransition = MathUtilsExt.SmoothInOutLerpFloat(positionTransitionAmount);
+                m_CanvasGroup.alpha = Mathf.Lerp(currentAlpha, targetAlpha, alphaSmoothTransition);
+                textTransform.localPosition = Vector3.Lerp(textCurrentLocalPosition, textTargetLocalPosition, positionSmoothTransition);
+                alphaTransitionAmount += Time.deltaTime * transitionSubtractMultiplier;
+                positionTransitionAmount += alphaTransitionAmount * 1.35f;
+                yield return null;
+            }
+
+            textTransform.localPosition = textTargetLocalPosition;
+            m_CanvasGroup.alpha = targetAlpha;
+            m_VisibilityCoroutine = null;
+
+            if (!fadeIn)
```

Comment:
Destroying this object seems like a pretty big/important side effect of not fading in